c7n/resources/awslambda.py

```diff
@@ -142,7 +183,98 @@ def _augment(r):
             resources, event)
 
 
-@AWSLambda.action_registry.register('delete')
+@actions.register('mark-for-op')
+class TagDelayedAction(TagDelayedAction):
+    """Action to specify an action to occur at a later date
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: lambda-delete-unused
+                resource: lambda
+                filters:
+                  - "tag:custodian_cleanup": absent
+                actions:
+                  - type: mark-for-op
+                    tag: custodian_cleanup
+                    msg: "Unused lambda"
+                    op: delete
+                    days: 7
+    """
+
+    permissions = ('lambda:TagResource',)
+
+    def process_resource_set(self, functions, tags):
+        client = local_session(self.manager.session_factory).client('lambda')
+        tag_dict = {}
+        for t in tags:
+            tag_dict[t['Key']] = t['Value']
+        for f in functions:
+            arn = f['FunctionArn']
+            client.tag_resource(Resource=arn, Tags=tag_dict)
+
+
+@actions.register('tag')
+class Tag(Tag):
+    """Action to add tag(s) to Lambda Function(s)
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: lambda-add-owner-tag
+                resource: lambda
+                filters:
+                  - "tag:OwnerName": missing
+                actions:
+                  - type: tag
+                    key: OwnerName
+                    value: OwnerName
+    """
+
+    permissions = ('lambda:TagResource',)
+
+    def process_resource_set(self, functions, tags):
+        client = local_session(self.manager.session_factory).client('lambda')
```

Comment:
this is basically a duplicate block, extracting to a helper function and invoking from both call sites would be preferred to avoid the duplication.